File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    num_list = {
        'zero': "0", 
        'one' : "1", 
        'two' : "2", 
        'three' : "3", 
        'four' : "4", 
        'five' : "5", 
        'six' : "6", 
        'seven' : "7", 
        'eight' : "8", 
        'nine' : "9",
    }
    
    answer = ''
    
    word = '' #현재 읽고 있는 단어 저장 변수
    for char in s:
        if char.isalpha():
            word += char
            if word in num_list:
                answer += num_list[word]
                word = "" #초기화
        else:
            answer += char
    
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    var numbers = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
    
    
    numbers.forEach((num, i)=>{
        while(s.includes(num)) s = s.replace(num, i);
    });
 
    return +s;
}